prov/gni/src/gnix_msg.c

```diff
@@ -1458,47 +2248,106 @@ static int _gnix_send_req(void *arg)
 	assert(rc == FI_SUCCESS);
 
 	if (unlikely(rendezvous)) {
-		assert(req->msg.send_md);
-
-		tag = GNIX_SMSG_T_RNDZV_START;
-		tdesc->rndzv_start_hdr.flags = req->msg.send_flags;
-		tdesc->rndzv_start_hdr.imm = req->msg.imm;
-		tdesc->rndzv_start_hdr.msg_tag = req->msg.tag;
-		tdesc->rndzv_start_hdr.mdh = req->msg.send_md->mem_hndl;
-		tdesc->rndzv_start_hdr.addr = req->msg.send_addr;
-		tdesc->rndzv_start_hdr.len = req->msg.send_len;
-		tdesc->rndzv_start_hdr.req_addr = (uint64_t)req;
-
-		if (req->msg.send_addr & GNI_READ_ALIGN_MASK) {
-			tdesc->rndzv_start_hdr.head =
-				*(uint32_t *)(req->msg.send_addr &
-					      ~GNI_READ_ALIGN_MASK);
-			GNIX_INFO(FI_LOG_EP_DATA,
-				  "Sending %d unaligned head bytes (%x)\n",
-				  GNI_READ_ALIGN -
-				  (req->msg.send_addr & GNI_READ_ALIGN_MASK),
-				  tdesc->rndzv_start_hdr.head);
-		}
+		if (req->msg.send_flags & GNIX_MSG_IOV) {
```

Comment:
You could have also used the request type to determine if you had a vectored send.  I think you used the last available flag bit for this.